test/com/google/enterprise/adaptor/documentum/DocumentumAdaptorTest.java

```diff
@@ -1995,52 +1999,57 @@ private void testDocContent(Date lastCrawled, Date lastModified,
     String content = "<html><body>Hello</body></html>";
     insertDocument(lastModified, path, mimeType, content);
 
-    MockResponse response = getDocContent(ImmutableMap.<String, String>of(),
+    RecordingResponse response = getDocContent(ImmutableMap.<String, String>of(),
         new MockRequest(docIdFromPath(path, DOCUMENT.pad("aaa")),
             lastCrawled));
 
     assertDocContent(response, lastModified, mimeType, content,
         expectNoContent);
   }
 
-  private void assertDocContent(MockResponse response, Date lastModified,
+
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  private void assertDocContent(RecordingResponse response, Date lastModified,
       String mimeType, String content, boolean expectNoContent)
       throws IOException {
     // Our mocks, like Documentum, only store seconds, not milliseconds.
     assertEquals(new Date((lastModified.getTime() / 1000) * 1000),
-        response.lastModified);
-    assertFalse(response.notModified);
-    assertFalse(response.metadata.isEmpty());
-    assertNotNull(response.acl);
+        response.getLastModified());
+    assertFalse(response.getState() == State.NOT_MODIFIED);
+    assertFalse(response.getMetadata().isEmpty());
+    assertNotNull(response.getAcl());
     if (expectNoContent) {
-      assertTrue(response.noContent);
-      assertEquals(null, response.contentType);
-      assertEquals(null, response.content);
+      assertTrue(response.getState() == State.NO_CONTENT);
+      assertEquals(null, response.getContentType());
+      exception.expect(IllegalStateException.class);
+      assertEquals(null, response.getOutputStream());
     } else {
-      assertFalse(response.noContent);
-      assertEquals(mimeType, response.contentType);
-      assertEquals(content, response.content.toString(UTF_8.name()));
+      assertFalse(response.getState() == State.NO_CONTENT);
```

Comment:
Use assertEquals with SEND_BODY or whatever, or at least add a message argument with the actual state.